media/js/base/mozilla-monitor-button.js

```diff
@@ -20,17 +20,27 @@ Mozilla.MonitorButton = (function() {
     }
 
     var buttonURL = monitorButton.getAttribute('href');
+    // strip url to everything after `?`
+    var buttonURLParams = buttonURL.match(/\?(.*)/)[1];
+
     var destURL = monitorButton.getAttribute('data-action') + 'metrics-flow';
 
-    fetch(destURL).then(function(resp) {
+    // collect values from monitor button
+    var params = window._SearchParams.queryStringToObject(buttonURLParams);
+
+    // add required params to the token fetch request
+    destURL += '&' + params.utm_source;
```

Comment:
Don't you still need to do this for the other params in the issue as well?

Reply:
ah jeeze. 🤦‍♀ whoops.